src/vm/comdelegate.cpp

```diff
@@ -352,6 +378,10 @@ VOID GenerateShuffleArray(MethodDesc* pInvoke, MethodDesc *pTargetMeth, SArray<S
     ArgLocDesc sArgSrc;
     ArgLocDesc sArgDst;
 
+#if defined(UNIX_AMD64_ABI) && defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+    int argSlots = NUM_FLOAT_ARGUMENT_REGISTERS + NUM_ARGUMENT_REGISTERS + sArgPlacerSrc.SizeOfArgStack() / sizeof(size_t);
```

Comment:
Nit: Can this be inside the big ifdef below?

Reply:
Unfortunately it cannot. The sArgPlacerSrc.SizeOfArgStack() cannot be called after the iteration has finished. It asserts due to that if I try to call it at that place, that's why I had to move it here.

Reply:
ok